### ISSUE TITLE
Plans to support OpenSearch RestClient?

### ISSUE THREAD

**Issue by sameer-b** [senacor/elasticsearch-evolution#198]
Hello,
Thanks for your hard work on this library! I'm trying to use it with OpenSearch, for now the ElasticSearch RestClient should work fine. But I'm curious if you have any plans to support the [OpenSearch RestClient](https://github.com/opensearch-project/OpenSearch/blob/main/client/rest/src/main/java/org/opensearch/client/RestClient.java)?

**Comment by S1riU5** [comment]
Hey @sameer-b,

not sure if there is any plan to incorporate the OpenSearch rest client directly, but maybe it makes sense to offer an api to provide a custom client and configure the client inside your application. I can create a PR  with an initial idea as a basis for discussions. @xtermi2 feel free to decline it if it does not fit into the overall goals.

**Comment by sameer-b** [comment]
@S1riU5 That would be ideal! Since we are moving to opensearch from elasticsearch our goal is to avoid any ES dependency. I'm guessing this will be a common pattern we will see. If the library is flexible to accept either, it will be great! thank you!

**Comment by xtermi2** [comment]
@S1riU5 hey you are stil alive. Nice to hear from you 😀

I see 2 possible solutions:
1. Build an abstraction over the ES rest client and implement in 2 modules a Elasticsearch and OpenSearch Adapter.
  * There is also an option to implement an adapter to use any other HTTP client like ApacheHTTP or OKHttp client.
2. Migrate from ES Rest Client to ApacheHTTP client or OKHttp client.

At the moment I would tend to option 2.

What do you think?

**Comment by S1riU5** [comment]
@xtermi2 Yes I am :-) also nice to hear from you. Initially, I was in favor of option 1, but solution 2 should also do the trick and should be backward compatible..... I'll implement version 2, to get rid of the Elasticsearch dependency, and look afterward into solution 1.  Thanks for your input!

**Comment by iceman91176** [comment]
Hey - we would also be interested in this. We would like to use OpenSearch Restclient in our applikcation with spring-data-opensearch (https://github.com/opensearch-project/spring-data-opensearch). Using the opensearch-client seems to be the recommended way, but means to exclude the Rest-Client (so using elaticsearch-evolution is not possible)

**Comment by iceman91176** [comment]
@S1riU5 - do you have any idea when (or if at all) this will be implemented? 
Best regards

**Comment by S1riU5** [comment]
@iceman91176  I'm sorry. At the moment, I don't have a lot of time to look into it but feel free to create a PR. :-) Sorry for the inconvenience.
